### PR TITLE
Relationship bug

### LINKED ISSUE #135: Relationship creation not working
This is due to using `FlaskForm`.

### PR DESCRIPTION
**Pull request by rmarow**
resolves #135

### CODE REVIEW

File: usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html [hunk 1]
```diff
@@ -68,12 +67,11 @@ <h4>Societal Benefit Areas</h4>
 
     <h4>Relationship</h4>
     {% if sqla_inspect(relationship).transient %}
-      {{ application_societal_benefit_area_fields(form.relationship.form) }}
+      {{ render_form(form.relationship.form) }}
```
Comment (rmarow) [inline usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html:70]: Wondering if there is a way to alter the submit button to say "create new relationship"

Comment (MattF-NSIDC) [reply usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html:70]: Does `button_map` support it?

**Review by MattF-NSIDC — CHANGES_REQUESTED** [review]
I think there are a good number of macros we don't use anymore that can be deleted, right?

File: usaon_vta_survey/templates/response/applications.html [hunk 2]
```diff
@@ -56,10 +56,8 @@ <h3>Applications</h3>
 
   <h3>Add application</h3>
   <form method="POST">
-    {{form.csrf_token}}
-    {{application_fields(form)}}
+    {{render_form(form, button_map={'submit_button': 'primary'}) }}
```
[the remark this answers is not in the document]
Comment (rmarow) [reply usaon_vta_survey/templates/response/applications.html:59]: I removed it and it makes nothing different! good call.

File: usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html (hunk 1 above)
Comment (MattF-NSIDC) [reply usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html:70]: Looked at the docs, and no, `button_map` is only for styles. I think this needs to be set in the `SubmitField` object? Either way, I think we should open an issue for this and move on :)

File: usaon_vta_survey/routes/response/relationships/data_product_application.py [hunk 3]
```diff
@@ -135,14 +141,17 @@ def view_response_data_product_application_relationships(survey_id: str):
...
+    class ResponseDataProductApplicationForm(SuperForm):
         """Combine all necessary forms into one super-form.
 
         NOTE: Additional class attributes are added dynamically below.
         """
 
         relationship = FormField(FORMS_BY_MODEL[ResponseDataProductApplication])
 
+    # looking for submit button at this point
+    # breakpoint()
```
Comment (rmarow) [inline usaon_vta_survey/routes/response/relationships/data_product_application.py:154]: At this point there is a submit button on `dir(SuperForm)`

File: usaon_vta_survey/routes/response/relationships/data_product_application.py [hunk 4]
```diff
@@ -168,6 +177,7 @@ class SuperForm(FlaskForm):
         data_product_id=data_product_id,
         application_id=application_id,
     )
+    # breakpoint()
```
Comment (rmarow) [inline usaon_vta_survey/routes/response/relationships/data_product_application.py:180]: observations at this point:
```>>> dir(ResponseDataProductApplicationForm)
['Meta', '__annotations__', '__class__', '__contains__', '__delattr__', '__delitem__', '__dict__', '__dir__', '__doc__', '__eq__', '__format__', '__ge__', '__getattribute__', '__getitem__', '__gt__', '__hash__', '__init__', '__init_subclass__', '__iter__', '__le__', '__lt__', '__module__', '__ne__', '__new__', '__reduce__', '__reduce_ex__', '__repr__', '__setattr__', '__setitem__', '__sizeof__', '__str__', '__subclasshook__', '__weakref__', '_cleanup_submit_buttons', '_unbound_fields', '_wtforms_meta', 'data', 'data_product', 'errors', 'hidden_tag', 'is_submitted', 'populate_obj', 'process', 'relationship', 'subforms', 'submit_button', 'validate', 'validate_on_submit'  ]```

File: usaon_vta_survey/routes/response/relationships/data_product_application.py [hunk 5]
```diff
@@ -205,9 +216,12 @@ class SuperForm(FlaskForm):
...
             db.session.commit()
 
-        return redirect(url_for('view_response_applications', survey_id=survey.id))
+        return redirect(
+            url_for('application.view_response_applications', survey_id=survey.id)
+        )
 
-    form = SuperForm(obj=form_obj)
+    form = ResponseDataProductApplicationForm(obj=form_obj)
+    # breakpoint()
```
Comment (rmarow) [inline usaon_vta_survey/routes/response/relationships/data_product_application.py:224]: Screen Shot 2023-09-28 at 3 14 03 PM
Screen Shot 2023-09-28 at 3 14 31 PM

File: usaon_vta_survey/routes/response/relationships/data_product_application.py [hunk 6]
```diff
@@ -180,8 +190,9 @@ class SuperForm(FlaskForm):
     }
```
Comment (rmarow) [inline usaon_vta_survey/routes/response/relationships/data_product_application.py:191]: still seeing `submit_field` on `ResponseDataProductApplicationForm` here...

File: usaon_vta_survey/__init__.py [hunk 7]
```diff
@@ -85,11 +94,14 @@ def before_request():
     app.register_blueprint(login_bp)
     app.register_blueprint(logout_bp)
     app.register_blueprint(blueprint, url_prefix="/google_oauth")
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/__init__.py:96]: Hate to nitpick on one more... should this be `google_bp` or `google_login_bp` for consistency?

File: usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html [hunk 8]
```diff
@@ -49,31 +48,8 @@ <h4>Societal Benefit Areas</h4>
...
-    </div>
-
-    <h4>Relationship</h4>
-    {% if sqla_inspect(relationship).transient %}
-      {{ application_societal_benefit_area_fields(form.relationship.form) }}
-    {% else %}
-      {{relationship}}
-    {% endif %}
-
-    <input type="submit" value="Create new relationship" />
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html:76]: Love seeing all these deletions :)

Comment (rmarow) [reply usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html:76]: @MattF-NSIDC it's still showing changes requested, but I think they're all covered. Let me know if there's anything else?

File: usaon_vta_survey/util/superform.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+    def subforms(self) -> list:
+        subforms_dict = {
+            key: field
+            for key, field in self._fields.items()
+            if isinstance(field, FormField) and 'relationship' not in key
+        }
+        subforms = list(subforms_dict.values())
+        # subforms = [
+        #     field for field in self._fields.values() if isinstance(field, FormField)
+        # ]
```
Comment (MattF-NSIDC) [inline usaon_vta_survey/util/superform.py:30]: Cleanup old code

**Review by MattF-NSIDC — COMMENTED** [review]
Can you summarize what the fix to the issue with the forms was in the end?

File: usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html (hunk 8 above)
Comment (MattF-NSIDC) [reply usaon_vta_survey/templates/response/relationships/application_societal_benefit_area.html:76]: If you click the refresh icon next to the "changes requested" red icon, you can clear it.

**Review by MattF-NSIDC — APPROVED** [review]
